src/client/src/components/pages/participants/participantDates/index.jsx

```diff
@@ -4,81 +4,121 @@ import Header from "../../../abstract/header";
-        "mohannadhanafi",
-        "15/5/2017",
-        "15/12/2017",
-        <React.Fragment>
-          <i className="fas fa-trash-alt" />
-          <Link to="/participants/date/details">
-            <i className="fas fa-info-circle" />
-          </Link>
-        </React.Fragment>
-      ]
-    ]
+    rows: [],
+    message: ""
   };
+
   onChange = event => {
     const { value, name } = event.target;
     this.setState({ [name]: value });
   };
+
   goAdd = event => {
     this.props.history.push("/participants/date/add");
   };
+
   goBack = event => {
-    this.props.history.push("/participant/details");
+    const id = this.props.match.params.id;
+    this.props.history.push(`/participant/details/${id}`);
+  };
+
+  // axios to make requests from backend.. 
+  getDates = async () => {
+    const { dispatch } = this.props.context;
+    const id = this.props.match.params.id;
+    try {
+      const result = await axios(`/api/v2/participant/${id}/dates`);
+      const dates = result.data.participantDates;
+      let array = [["Participant Name/BB-No", "Worker Name/ID", "Start", "End", "Action"]];
+      if (dates.length === 0){
+        const msg = 'There is no dates yet !!';
+        array =[];          
+        this.setState({ message: msg, rows:array});
+      }
+      else{    
+      dates.map(row =>
+        array.push([
+          row.participant_id,
+          row.worker_id,
+          row.date_start.split("T")[0],
+          row.date_end.split("T")[0],
+          <>
+             <i className="fas fa-trash-alt"  onClick={() => this.onDelete(row.id)}/>
+             <Link to= {`/participant/${id}/date/details/${row.id}`}>
+               <i className="fas fa-info-circle" />
+             </Link>
+           </>
+         ])
+       );
+       this.setState({ rows: array });
+       }
+    } catch (err) {
+      dispatch({ type: 'ERROR_PAGE', payload: { ErrorPage: err.response.status } })
+    }
+};
+
+  componentDidMount = async () => {
+    this.getDates();
   };
+
+//Delete an exist date
+  onDelete = id =>{
+    swal({
+      type: 'warning',
+      html:'Are you sure that you want to delete this date ?',
+      showCancelButton: true,
+      focusConfirm: false,
+      confirmButtonText:'<i class="fa fa-thumbs-up"></i> Yes',
+      confirmButtonAriaLabel: 'Thumbs up',
+      cancelButtonText:'<i class="fa fa-thumbs-down"></i> No ',
+      cancelButtonAriaLabel: 'Thumbs down',
+    }).then(confirm => {
+      if (confirm.value) {
+        axios("/api/v2/date", {
+          method: "DELETE",
+          data: {
+            dateId: id
+          }
+        }).then(result => {
+          this.getDates().then(() => {
+            swal({
+              title: 'Success',
+              type: 'success',
+              html: ' <strong>Your work has been saved</strong> <br/>' +result.data.message,
+              showConfirmButton: false,
+              timer: 3000
+            }).then(() => {
+              this.getDates();
```

Comment:
I think you only need one call to getDates() here (so you could remove the one on line 98, after the sweetalert).  Since you are waiting for the DELETE axios call to finish, your getDates() on line 90 is guaranteed to get the correct list of dates (i.e. it will reflect the fact that one was just deleted).
Also, you don't need to wait for the `getDates()` to finish to do the `swal()` call inside the `then()`.  Instead, you could just do them both at the same time:
```
this.getDates()
swal({
  ...
})
```
That will save the user a little bit of waiting 😀